pages/contact.tsx

```diff
@@ -71,9 +75,31 @@ export default function ContactPage() {
 
     if (errors.length) {
       setFormError(`Please correct the following issues before submitting: ${errors.join(', ')}`);
-    } else {
-      setFormError('');
-      (e.target as HTMLFormElement).submit();
+      return;
+    }
+
+    try {
+      const response = await fetch(API_ENDPOINT, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(toSend),
+      });
+
+      if (response.ok) {
+        setFormError('');
+        setFormSuccess('Your message was sent successfully!');
+      } else {
+        const errorText = await response.text();
+        setFormError(`Server error: ${errorText}`);
+      }
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        setFormError(`Network error: ${error.message}`);
+      } else {
+        setFormError('An unknown error occurred');
+      }
     }
```

Comment:
Hey @emilyemilyemilyemilyemilyemily
Can I suggest one thing here? 
This logic that handle our request doesnt seems to be a `contact.tsx` page`s responsible.

We are adding logic here that increase our complexity.
What do you think about using a `custom hook` to help us to separate that request logic and just use `methods` and `properties` that are exposed by that hook?

I created an example to make this more clear [here](https://gist.github.com/LeoFuna/f85be3cb21ac57a08cd007017b5ee052)
```suggestion
    makeRequest(toSend);
```